src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -11,27 +11,468 @@
 -----------------------------------------------------------------------------
 module Algebra.Graph.Relation.Symmetric (
     -- * Data structure
-    SymmetricRelation, fromRelation, toRelation,
+    SymmetricRelation(..), fromRelation, toRelation,
+
+    -- * Basic graph construction primitives
+    empty, vertex, edge, overlay, connect, vertices, edges, overlays, connects,
+
+    -- * Relations on graphs
+    isSubgraphOf,
 
     -- * Graph properties
-    neighbours
+    isEmpty, hasVertex, hasEdge, vertexCount, edgeCount, vertexList, edgeList,
+    adjacencyList, vertexSet, edgeSet, neighbours,
+
+    -- * Standard families of graphs
+    path, circuit, clique, biclique, star, stars, tree, forest,
+
+    -- * Graph transformation
+    removeVertex, removeEdge, replaceVertex, mergeVertices, gmap, induce,
+
+    -- * Relational operations
+    compose
   ) where
 
-import Algebra.Graph.Relation
-import Algebra.Graph.Relation.InternalDerived
+import qualified Algebra.Graph.Relation as R
+import qualified Algebra.Graph.Relation.Internal as RI hiding (empty, vertex, overlay, connect, referredToVertexSet)
+import Algebra.Graph.Relation.Symmetric.Internal
 
 import qualified Data.Set as Set
+import Data.Tree 
+import qualified Data.Tree as Tree
+import Data.Tuple (swap)
 
 -- | Construct a symmetric relation from a 'Relation'.
 -- Complexity: /O(1)/ time.
-fromRelation :: Relation a -> SymmetricRelation a
+fromRelation :: R.Relation a -> SymmetricRelation a
 fromRelation = SymmetricRelation
 
 -- | Extract the underlying relation.
 -- Complexity: /O(m*log(m))/ time.
-toRelation :: Ord a => SymmetricRelation a -> Relation a
-toRelation = symmetricClosure . fromSymmetric
+toRelation :: Ord a => SymmetricRelation a -> R.Relation a
+toRelation = R.symmetricClosure . fromSymmetric
+
+-- | Construct the graph comprising /a single edge/.
+-- Complexity: /O(1)/ time, memory and size.
+--
+-- @
+-- edge x y                 == edges [(x,y),(y,x)]
```

Comment:
This doesn't appear to be tested. All new examples should be added to the testsuite.

Reply:
Also, I think we should keep this example too: `edge x y == connect (vertex x) (vertex y)`.

Reply:
Actually I'm going to remove this one because it's innacurate, the show instance does not show the repeated edges.

Reply:
The property `edge x y == edges [(x,y),(y,x)]` is correct: it doesn't involve `show` at all. So I actually think it's a nice example, but it should of course be tested, as all others.

Reply:
Well if I type that on the repl it does not output. I actually followed your recommendation on this so in the show instance I filter the repeated edges.

![image](https://user-images.githubusercontent.com/20261421/55432183-25d3dc00-558a-11e9-8180-1998bb497390.png)

Shouldn't I have done it? I can easily fix the show instance.

Reply:
It only now crossed my mind it's the `edges` function. I was too hung up on this issue.

Reply:
`edge x y` and `edges [(x,y),(y,x)]` correspond to the same undirected graph which contains an edge between vertices `x` and `y`.

I don't understand how our discussion about `edgeCount` is related to this example. Both `edge x y` and `edges [(x,y),(y,x)]` have one edge.

Reply:
Yes, sorry to bring that up, it was to show you what motivated me to filter the repeated edges on the show instance.